### PR TITLE
[FEATURE] Update extension for TYPO3 Version 12

### PR DESCRIPTION
**Pull request by jonashirschi**
This commit updates the extension to be compatible with TYPO3 version 12. The extension is now on version 3.0.0 and contains the necessary changes to work seamlessly with the latest TYPO3 version, as well as with the previous versions 10 and 11. (currently still untested on TYPO3 10)

References #18

### CODE REVIEW

File: Classes/LinkHandler/PageLinkHandler.php [hunk 1]
```diff
@@ -147,17 +159,36 @@ protected function getRecordsOnExpandedPage($pageId)
                     )
                 )
                 ->orderBy('colPos')
-                ->addOrderBy('sorting')
-                ->execute()
-                ->fetchAll();
+                ->addOrderBy('sorting');
+
+            $versionInformation = GeneralUtility::makeInstance(Typo3Version::class);
```
Comment (brotkrueml) [inline Classes/LinkHandler/PageLinkHandler.php]: Above in PageLinkHandler `new Typo3Version()` is used, here GU:mI(). As this makes in practise no difference, I would suggest to use just one variant in the same code base.

File: Classes/LinkHandler/PageLinkHandler.php [hunk 2]
```diff
@@ -147,17 +159,36 @@ protected function getRecordsOnExpandedPage($pageId)
...
+            } else {
+                $contentElements = $contentElements
+                    ->executeQuery()
+                    ->fetchAllAssociative();
+            }
 
             $colPosArray = GeneralUtility::callUserFunction(BackendLayoutView::class . '->getColPosListItemsParsed', $pageId, $this);
             $languages = GeneralUtility::makeInstance(TranslationConfigurationProvider::class)->getSystemLanguages($pageId);
 
+            $versionInformation = GeneralUtility::makeInstance(Typo3Version::class);
```
Comment (brotkrueml) [inline Classes/LinkHandler/PageLinkHandler.php]: You have already `$versionInformation` above, so no need to get it again.

File: Classes/LinkHandler/PageLinkHandler.php [hunk 3]
```diff
@@ -147,17 +159,36 @@ protected function getRecordsOnExpandedPage($pageId)
...
                 )
                 ->orderBy('colPos')
-                ->addOrderBy('sorting')
-                ->execute()
-                ->fetchAll();
+                ->addOrderBy('sorting');
+
+            $versionInformation = GeneralUtility::makeInstance(Typo3Version::class);
+            if($versionInformation->getMajorVersion() < 12) {
+                $contentElements = $contentElements
```
Comment (brotkrueml) [inline Classes/LinkHandler/PageLinkHandler.php]: Personally, I don't like to "hijack" the result variable name from the query builder and assign it anew, which is then an array. I would prefer to use a different variable name. Also some lines below.

**Review by brotkrueml — COMMENTED** [review]
Thank you for the PR, as I am just on upgrading to v12 this is really helpful (and works for me) :+1:

**Comment by georgringer** [comment]
works for me. TBH I would just drop v10 support - no need for that

**Comment by bmack** [comment]
Yes. @jonashirschi can you update the PR?

**Comment by ochorocho** [comment]
Fixed with https://github.com/b13/link2language/pull/23 in v3.0.0

**Pull request closed without merging**